clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,11 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            if (!this.subscriptions.allConsumed().isEmpty() && this.autoCommitEnabled)
```

Comment:
Interesting. So this seems to expose a couple problems. 

1. It looks like we allow the `OffsetCommitCallback` provided by the user to be null, so we probably need a null check in `ConsumerCoordinator.OffsetCommitCompletion.invoke()`.
2. For automatic commits, we use a logging `OffsetCommitCallback`. See `ConsumerCoordinator.maybeAutoCommitOffsetsAsync()`. We should probably use the same callback here, so maybe we need to make that method public  and invoke it instead of calling `commitAsync`.